progress/progress.go

```diff
@@ -0,0 +1,94 @@
+package progress
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"runtime/debug"
+	"sort"
+	"sync/atomic"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/livepeer/catalyst-api/clients"
+	"github.com/livepeer/catalyst-api/log"
+)
+
+var Clock = clock.New()
+
+var progressReportBuckets = []float64{0, 0.25, 0.5, 0.75, 1}
+
+const minProgressReportInterval = 10 * time.Second
+const progressCheckInterval = 1 * time.Second
+
+func ReportProgress(ctx context.Context, client clients.CallbackClient, url, taskID string, size uint64, getCount func() uint64, startFraction, endFraction float64) {
```

Comment:
Nice! Btw this is a simpler implementation of the progress reporter. It has been refactored for a nicer interface which you can dynamically update the end fraction. I had sent the link to the right branch, but I guess you ended up basing this on `main`. If you wanna take a look on the new one, here it is: https://github.com/livepeer/task-runner/blob/c47a871f37f4df8e8bea72a0b7f6e97a1713b8e5/task/progress.go#L35

Reply:
@victorges, just changed this to the new reporter's implementation. Would you please mind to take a look? BTW, if we pass a callback instead of a client, we would decouple the package from the notifier logic, WDYT?

Reply:
Yeah I like the decoupling! Can even move it to `go-tools` if we deem appropriate 😄 (might get a little annoying for logging tho)